tools/ci/tests/test_run_tc.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+from six import iteritems
+
+from tools.ci import run_tc
+
+
+@pytest.mark.parametrize("msg,expected", [
+    ("Some initial line\n\ntc-jobs:foo,bar", set(["foo", "bar"])),
+    ("Some initial line\n\ntc-jobs:foo, bar", set(["foo", "bar"])),
+    ("tc-jobs:foo, bar   \nbaz", set(["foo", "bar"])),
+    ("tc-jobs:all", set(["all"])),
+    ("", set())])
+@pytest.mark.parametrize("event", [
+    {"commits": [{"message": "<message>"}]},
+    {"pull_request": {"body": "<message>"}}
+])
+def test_extra_jobs_pr(msg, expected, event):
+    def sub(obj):
```

Comment:
This took me a while to suss out. Would you mind just duplicating the first parameter set instead?

Reply:
That seems needlessly repetative and hard to maintain. I'll add a comment instead.

Reply:
I find recursive functions harder to maintain than lists of values, but to each his own.